src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -33,107 +40,153 @@
         produces = "application/json")
     public ResponseEntity<List<Pet>> importPets(@RequestBody String csv) {
 
-        int i = 0;
         List<Pet> pets = new LinkedList<Pet>();
         Pet pet;
 
-        do {
-            pet = new Pet();
+        List<String> splitCSV = segment_line_to_entries(csv);
```

Comment:
you could think about extracting this content to a new method to separate the responsibilities further